### PR TITLE
Bump up num-complex to 0.3.1

### PR DESCRIPTION
**Pull request by termoshtt**
- blas: https://github.com/blas-lapack-rs/blas/pull/34
- cblas: https://github.com/blas-lapack-rs/cblas/pull/4
- lapack: https://github.com/blas-lapack-rs/lapack/pull/25
- lapacke: https://github.com/blas-lapack-rs/lapacke/pull/3

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,12 +17,12 @@ categories = ["api-bindings", "science"]
 keywords = ["linear-algebra"]
 
 [dependencies]
-libc = "0.2"
+libc = "0.2.84"
```
Comment (IvanUkhov) [inline Cargo.toml:20]: Why is this needed?

File: Cargo.toml [hunk 2]
```diff
@@ -17,12 +17,12 @@ categories = ["api-bindings", "science"]
...
 
 [dependencies.num-complex]
-version = "0.2"
+version = "0.3.1"
```
Comment (IvanUkhov) [inline Cargo.toml:23]: I propose to drop `.1`, or is there a reason to keep it?

File: Cargo.toml [hunk 3]
```diff
@@ -17,12 +17,12 @@ categories = ["api-bindings", "science"]
...
 default-features = false
 
 [dependencies.blas-sys]
-version = "0.7"
+version = "0.7.1"
```
Comment (IvanUkhov) [inline Cargo.toml:27]: Is this needed?

**Review by IvanUkhov — APPROVED** [review]

File: Cargo.toml (hunk 2 above)
Comment (termoshtt) [reply Cargo.toml:23]: This is not needed indeed since this crate works with num-complex 0.3.0. If you actually want to support all "0.3.x" versions of num-complex and other crates, I will drop all patch versions.

IMO, version specification without patch version, `num-complex = "0.3"` in this case, means this crate works with all versions of num-complex `0.3.x`, but it is generally too strict requirements comparing that `num-complex = "0.3.1"` means this crate works with num-complex `0.3.x` where `x >= 1`.  We usually test these crates with latest dependent crates, and cargo users can get latest crates easily. Thus I encourage to use full version specification like "0.3.1" always to avoid mysterious broken Cargo.toml

Comment (IvanUkhov) [reply Cargo.toml:23]: I think we should rely on semantic versioning here. Before version 1.0, there are no real promises, but people usually follow the convention that patch versions are backward compatible. Specifying a patch version is unnecessary constraint, which one would not want to have unless one knows that something new was added, and that something new is needed. I would stop at minor versions in all dependencies.

Comment (termoshtt) [reply Cargo.toml:23]: OK. I will remove them.
